src/Uno.UI/UI/Xaml/Controls/MenuFlyout/CascadingMenuHelper.cs

```diff
@@ -0,0 +1,976 @@
+
+				if (ownerAsDO != null)
+				{
+					ElementSoundPlayerService.RequestInteractionSoundForElementStatic(ElementSoundKind.Hide, ownerAsDO as DependencyObject);
+				}
+			}
+		}
+
+		internal void CloseChildSubMenus()
+		{
+			// WeakRefPtr fails an assert if we attempt to AsOrNull() to a type
+			// that we aren't sure if the contents of the weak reference
+			// implement that type.  To avoid that assert, we first As() the
+			// weak reference contents to a known type that it's guaranteed
+			// to be (if it isn't null), and we then AsOrNull() the ComPtr,
+			// once it's safe to ask about a type that we're not sure of.
+			FrameworkElement subMenuPresenterAsFE = m_wpSubMenuPresenter?.Target as Frame;
+
+			IMenuPresenter subMenuPresenter = null;
+
+			if (subMenuPresenterAsFE != null)
+			{
+				subMenuPresenter = subMenuPresenterAsFE as IMenuPresenter;
+			}
+
+#if CMH_DEBUG
+			(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: CloseChildSubMenus - subMenuPresenter=0x%p.", this, subMenuPresenter));
+#endif // CMH_DEBUG
+
+			if (subMenuPresenter != null)
+			{
+				subMenuPresenter.CloseSubMenu();
+			}
+		}
+
+		internal void DelayCloseSubMenu()
+		{
+			EnsureDelayCloseMenuTimer();
+			if (m_delayCloseMenuTimer != null)
+			{
+#if CMH_DEBUG
+				(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: DelayCloseSubMenu - Starting m_delayCloseMenuTimer.", this));
+#endif // CMH_DEBUG
+
+				m_delayCloseMenuTimer.Start();
+			}
+		}
+
+		internal void CancelCloseSubMenu()
+		{
+			if (m_delayCloseMenuTimer != null) 
+			{
+#if CMH_DEBUG
+				(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: CancelCloseSubMenu - Stopping m_delayCloseMenuTimer.", this));
+#endif // CMH_DEBUG
+
+				m_delayCloseMenuTimer.Stop();
+			}
+		}
+
+		internal void ClearStateFlags()
+		{
+#if CMH_DEBUG
+			(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: ClearStateFlags.", this));
+#endif // CMH_DEBUG
+
+			m_isPressed = false;
+			m_isPointerOver = false;
+			UpdateOwnerVisualState();
+		}
+
+		internal void OnIsEnabledChanged()
+		{
+			Control ownerAsControl = m_wpOwner?.Target as Control;
+
+#if CMH_DEBUG
+			(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnIsEnabledChanged - ownerAsControl=0x%p.", this, ownerAsControl));
+#endif // CMH_DEBUG
+
+			if (ownerAsControl != null)
+			{
+				bool bIsEnabled = false;
+				bIsEnabled = (ownerAsControl.IsEnabled);
+
+				if (!bIsEnabled)
+				{
+					ClearStateFlags();
+				}
+				else
+				{
+					ownerAsControl.UpdateVisualState(true /* useTransitions */);
+				}
+			}
+		}
+
+		public void OnVisibilityChanged()
+		{
+			UIElement ownerAsUIE = m_wpOwner?.Target as UIElement;
+
+#if CMH_DEBUG
+			(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnVisibilityChanged - ownerAsUIE=0x%p.", this, ownerAsUIE));
+#endif // CMH_DEBUG
+
+			if (ownerAsUIE != null)
+			{
+				Visibility visibility = Visibility.Collapsed;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'visibility'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)